### DISCUSSION TITLE
Handling plugins not supported by Nixvim and not in nixpkgs

### DISCUSSION THREAD

**Discussion by vendion** [nix-community/nixvim#1331] [Q&A] [answered]
I'm currently rewriting my current neovim config to use nixvim, and I have a few plugins that are not supported directly by nixvim, and doesn't seem to be in the nixpkgs repos either. I do see that nixvim does support also using something like Lazy, but [the docs for how to use it](https://nix-community.github.io/nixvim/plugins/lazy/plugins.html) are pretty vuage.

I tried doing something like the following but, in doing so results in a failure when I run home-manager:

```nix
programs.nixvim = {
  plugins = {
      lazy = {
        enable = true;
        plugins = [
          {
            name = "laytan/cloak.nvim";
            opts = ''
            {
              enabled = true,
              cloak_character = "*",
              highlight_group = "Comment",
              patterns = {
                {
                  file_pattern = {
                    ".env*",
                    "*.toml",
                  },
                  cloak_pattern = {
                    "=.+",
                    ":.+",
                    "-.+",
                  },
                },
              },
            }
            '';
          }
        ];
      };
  };
};
```

When I do a `home-manager switch` I get the following error:

```
error: The option `programs.nixvim.plugins.lazy.plugins."[definition 1-entry 1]".pkg' is used but not defined.
```

**Comment by traxys** [comment, marked as answer]
Well using `lazy` is not really in the nixvim model (it is supported, but not the main path we think about let's say).

We have two entries in the FAQ to do it the "main way":
- [not in nixvim](https://nix-community.github.io/nixvim/user-guide/faq.html#how-do-i-use-a-plugin-not-implemented-in-nixvim)
- [not in nixpkgs](https://nix-community.github.io/nixvim/user-guide/faq.html#how-do-i-use-a-plugin-not-packaged-in-nixpkgs)

If you still want to use `lazy` you would be able to set `pkg = pkgs.vimUtils.buildVimPlugin ....` I think.

**Reply by traxys** [reply]
Also don't hesitate to create an issue to add the plugin to nixvim, we often add new plugins to nixpkgs!

**Reply by vendion** [reply]
Does it matter if the plugin isn't already in nixpkgs? I ask as while working on migrating my neovim configs I have identifed a couple of plugins that is not supported by nixvim and not in nixpkgs.

**Reply by traxys** [reply]
We can't integrate in the nixvim repo plugins that are not in nixpkgs, but whenerver we have plugin request issues for such plugins we often create PRs to nixpkgs to add them.
